src/components/AnonymousChat/child/ChannelFollowButton.tsx

```diff
@@ -3,24 +3,31 @@ import React from 'react';
 import {Pressable, StyleSheet, Text, View} from 'react-native';
 
 import dimen from '../../../utils/dimen';
-import {colors} from '../../../utils/colors';
 import {fonts} from '../../../utils/fonts';
+import {COLORS} from '../../../utils/theme';
 
 interface Props {
   isFollowing: boolean;
   handleFollow?: () => void;
+  isAnonymousTab?: boolean;
 }
 
 const ChannelFollowButton = (props: Props) => {
-  const {isFollowing, handleFollow} = props;
+  const {isFollowing, handleFollow, isAnonymousTab = false} = props;
 
   return (
     <Pressable onPress={handleFollow}>
       <View style={styles.followContainer}>
         <View
-          style={[styles.containerFollowBtn, isFollowing ? styles.followingBtn : styles.followBtn]}>
+          style={[
+            styles.containerFollowBtn,
+            isFollowing ? styles.followingBtn(isAnonymousTab) : styles.followBtn(isAnonymousTab)
+          ]}>
           <Text
-            style={[styles.textFollowBtn, isFollowing ? styles.textFollowing : styles.textFollow]}>
+            style={[
+              styles.textFollowBtn,
+              isFollowing ? styles.textFollowing(isAnonymousTab) : styles.textFollow
+            ]}>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The destructuring of props now includes `isAnonymousTab`, and it's given a default value of `false`. This is a good practice as it ensures that `isAnonymousTab` will always have a value, preventing potential errors when it's used later in the code. The styles of the button and text are now also dependent on `isAnonymousTab`.

```diff
  const ChannelFollowButton = (props: Props) => {
-   const {isFollowing, handleFollow} = props;
+   const {isFollowing, handleFollow, isAnonymousTab = false} = props;

    return (
      <Pressable onPress={handleFollow}>
        <View style={styles.followContainer}>
          <View
           style={[
             styles.containerFollowBtn,
-            isFollowing ? styles.followingBtn : styles.followBtn
+            isFollowing ? styles.followingBtn(isAnonymousTab) : styles.followBtn(isAnonymousTab)
           ]}>
            <Text
             style={[
               styles.textFollowBtn,
-              isFollowing ? styles.textFollowing : styles.textFollow
+              isFollowing ? styles.textFollowing(isAnonymousTab) : styles.textFollow
             ]}>
             {isFollowing ? 'Following' : 'Follow'}
           </Text>
         </View>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->